### ISSUE TITLE
SAM fig2

### ISSUE THREAD

**Issue by Jacobxz** [CHENGY12/APNet#6]
作者您好，感谢分享，我有几个问题想请教一下：

1、在SAM中，splite 切分的时候是切了通道数，`splited = torch.split(x, channel//self.radix, dim=1)`，而图2画的是将上下切分，或者切4份？

2、切分完，做注意力，2层的时候，level=2注意力只是进行了softmax处理，`atten = F.softmax(atten, dim=1).view(batch, -1, 1, 1)`，level=1 注意力用了SEnet？

3、按照图2，先全局，后逐步细粒度，但是代码里为啥顺序是相反的？
        x = self.base_1(x)
        if self.level > 2:
            x = self.att_ss1(x)
            x = self.BN_1(x)
        if self.level > 1:
            x = self.att_s1(x)
            x = self.BN1(x)
        if self.level > 0:
            y = self.att1(x)
            x = x*y.expand_as(x)

麻烦了~

**Comment by CHENGY12** [comment]
非常感谢关注我们的工作，
1） APNet 是可以基于任何形式的attention的， 我们在实验中给出了APNet-C （channel-wise attention）和APNet-S（spatial attention。图2里面描述的是基于spatial attention的APNet-S， 'splited = torch.split(x, channel//self.radix, dim=1)'  是APNet-C 对应的代码。 文章中有更详细的解释~

2） APNet 是对注意力网络的扩展， 当level=1 的时候， 它相当与正常的注意力网络， 对于APNet-C 来说，APNet-C （level=1）就是 SElayer.  对于 level>=2 的时候， 我们会对注意力做金字塔操作，split-attend-merge-stack ，其中attend 就是做原始的注意力操作， ‘atten = F.softmax(atten, dim=1).view(batch, -1, 1, 1)’ 是注意力代码的一部分， 也就是SElayer 的一部分

3）APNet 对于croase to fine 和fine to croase 是鲁棒的， 不过最后实验结果是基于code 中的fine to coarse得到的。我们在issue 4 中关于这个问题给了详细的讨论， 希望对你的问题有帮助。

**Comment by Jacobxz** [comment]
感谢感谢，明白了~能不能分享一下APNet-S（spatial attention）对应的代码，如果方便的话，202516180@qq.com，🙏

**Comment by trouble-killer** [comment]
Hi,
Thanks for sharing the code！
Would you mind sharing the code of spatial attention(APNet-S), and this is my email address if it's convenient，1127163106@qq.com
